### PR TITLE
update fuse container as root user in sidecar mode

### PR DESCRIPTION
**Pull request by zwwhdls**
Signed-off-by: zwwhdls <zww@hdls.me>

### Ⅰ. Describe what this PR does
fix https://github.com/fluid-cloudnative/fluid/issues/2500

### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: charts/juicefs/templates/fuse/daemonset.yaml [hunk 1]
```diff
@@ -127,8 +127,9 @@ spec:
           - containerPort: 9567
             name: metrics
             protocol: TCP
-          {{- if .Values.fuse.privileged }}
           securityContext:
+            runAsUser: 0
```
Comment (cheyang) [inline charts/juicefs/templates/fuse/daemonset.yaml:131]: Will you consider about `configurable nonRoot`  like https://github.com/fluid-cloudnative/fluid/blob/master/docs/zh/samples/nonroot_access.md ?

```
runAs:
    uid: 1201
    gid: 1201
    user: fluid-user-1
    group: fluid-user-1
```

Comment (zwwhdls) [reply charts/juicefs/templates/fuse/daemonset.yaml:131]: It's different from Alluxio, JuiceFS needs FUSE and only root user can access FUSE. NonRoot users can access data in JuiceFS by setting `runAs` in application pod.

Comment (cheyang) [reply charts/juicefs/templates/fuse/daemonset.yaml:131]: Got it. thanks.

**Comment by cheyang** [comment]
@zwwhdls could you update chart version？

**Comment by zwwhdls** [comment]
> 

done

**Pull request merged**